### ISSUE TITLE
Samples don't accurately reflect the computational results

### ISSUE THREAD

**Issue by arielsvn** [AlexsLemonade/refinebio#2064]
### Context

https://github.com/AlexsLemonade/refinebio/issues/2054#issuecomment-573110562

### Problem or idea

Some samples (processed or not) don't accurately reflect the computational results that produced their computed files.

The sample [ERR1663012](https://www.refine.bio/samples/ERR1663012) from the experiment [ERP017365](https://www.refine.bio/experiments/ERP017365/impc-high-throughput-rna-seq-of-1135-lines) is processed and has the following computational results.

```
'Salmon Quant'
time_start: 2019-3-28
'salmon --version': 'salmon 0.9.1'
'organism_index__assembly_name': 'GRCm38'

'Salmontools'
time_start: 2019-3-28
'salmontools --version': 'Salmon Tools 0.1.0'

'MultiQC'
time_start: 2019-3-28
'salmon --version': 'salmon 0.9.1', 
'fastqc --version': 'FastQC v0.11.7'

'Salmon Quant', 
time_start: 2019-9-9
'salmon --version': 'salmon 0.13.1', 
'sra-stat --version': 'sra-stat : 2.9.1' 
'organism_index__assembly_name': 'GRCm38' 

'Salmontools'
time_start: 2019-9-9
'salmontools --version': 'Salmon Tools 0.1.0'

'Tximport'
time_start: 2019-10-3
```

In the frontend we ignore repeated computational results, but if a sample when showing processing information, but it's hard to differentiate processed from unprocessed samples when both have processing information.

![image](https://user-images.githubusercontent.com/1882507/72173926-96741880-33a6-11ea-896d-858ed5fcbb7c.png)

### Solution or next step

I see two possible solutions:

- Have the frontend hide all processing information if a sample is not processed

- Make this modification in the API, and have the sample details endpoint only return valid computational results. In the case of the sample above it could only return the last salmon and tximport results. We could have a separate endpoint to list all computational results associated with a sample (@davidsmejia idea)

I lean towards the API modification but there might be other options. @kurtwheeler what do you think?

**Comment by arielsvn** [comment]
Raw query data from the computational results:

```
>>> Sample.objects.get(accession_code='ERR1663012').results.order_by('time_start').values('processor__name', 'time_start', 'organism_index__assembly_name', 'processor__environment__cmd_line')
<QuerySet [{'processor__name': 'Salmon Quant', 'processor__environment__cmd_line': {'salmon --version': 'salmon 0.9.1'}, 'time_start': datetime.datetime(2019, 3, 28, 20, 15, 35, 882594, tzinfo=<UTC>), 'organism_index__assembly_name': 'GRCm38'}, {'processor__name': 'Salmontools', 'processor__environment__cmd_line': {'salmontools --version': 'Salmon Tools 0.1.0'}, 'time_start': datetime.datetime(2019, 3, 28, 20, 20, 16, 880175, tzinfo=<UTC>), 'organism_index__assembly_name': None}, {'processor__name': 'MultiQC', 'processor__environment__cmd_line': {'salmon --version': 'salmon 0.9.1', '/home/user/FastQC/fastqc --version': 'FastQC v0.11.7'}, 'time_start': datetime.datetime(2019, 3, 28, 20, 20, 41, 676547, tzinfo=<UTC>), 'organism_index__assembly_name': None}, {'processor__name': 'Salmon Quant', 'processor__environment__cmd_line': {'salmon --version': 'salmon 0.13.1', 'sra-stat --version': 'sra-stat : 2.9.1'}, 'time_start': datetime.datetime(2019, 9, 9, 23, 50, 34, 803439, tzinfo=<UTC>), 'organism_index__assembly_name': 'GRCm38'}, {'processor__name': 'Salmontools', 'processor__environment__cmd_line': {'salmontools --version': 'Salmon Tools 0.1.0'}, 'time_start': datetime.datetime(2019, 9, 9, 23, 53, 42, 702712, tzinfo=<UTC>), 'organism_index__assembly_name': None}, {'processor__name': 'Tximport', 'processor__environment__cmd_line': None, 'time_start': datetime.datetime(2019, 10, 3, 21, 39, 32, 259368, tzinfo=<UTC>), 'organism_index__assembly_name': None}]>
```

**Comment by arielsvn** [comment]
I talked with @kurtwheeler, and we decided that the API should reflect what's on the database. This means that we'll have some samples that were processed with Salmon but they aren't downloadable because we haven't processed them with tximport.

@dvenprasad is something like this confusing for users? [SRP066781](https://www.refine.bio/experiments/SRP066781/single-cell-indexed-rna-seq-of-human-hematopoietic-stem-and-progenitors)

![image](https://user-images.githubusercontent.com/1882507/72183572-54ee6800-33bc-11ea-8071-2c93f996b94b.png)

We could also hide all processing information for unprocessed samples
